### ISSUE TITLE
Refuse to die

### ISSUE THREAD

**Issue by skinkie** [jiangwenyuan/nuster#81]
This one appeared again on master. I just upgraded master (5.0.0.21-2d7ef8-119) to see if it will fail again.
```
[NOTICE] 138/012000 (12161) : [nuster][cache] on, dict_size=1048576, data_size=1073741824
[WARNING] 138/012000 (12161) : [./haproxy.main()] Cannot raise FD limit to 2000047, limit is 4096. This will fail in >= v2.3
[NOTICE] 138/012000 (12161) : haproxy version is 5.0.0.21-77ecdb-115
[NOTICE] 138/012000 (12161) : path to executable is ./haproxy
[ALERT] 138/012000 (12161) : [./haproxy.main()] FD limit (4096) too low for maxconn=1000000/maxsock=2000047. Please raise 'ulimit-n' to 2000047 or more to avoid any trouble.This will fail in >= v2.3
[NOTICE] 138/012000 (12161) : New worker #1 (12163) forked
[ALERT] 141/114259 (12163) : A bogus STREAM [0x7fe25021c030] is spinning at 100249 calls per second and refuses to die, aborting now! Please report this error to developers [strm=0x7fe25021c030 src=83.84.14.59 fe=http be=openstreetmap_org dst=s1 rqf=48c4a000 rqa=40000 rpf=e0044060 rpa=24000000 sif=EST,208000 sib=CLO,280190 af=(nil),0 csf=0x7fe2541079b0,104000 ab=(nil),0 csb=0x7fe260192360,0 cof=0x7fe25820b550,1300:H2(0x7fe2542b9740)/SSL(0x7fe2601aee40)/tcpv4(42) cob=0x129d260,c2300:H1(0x11d9fc0)/RAW((nil))/tcpv6(159) filters={0x7fe268088570="cache filter id"}]
[ALERT] 141/114259 (12161) : Current worker #1 (12163) exited with code 134 (Aborted)
[ALERT] 141/114259 (12161) : exit-on-failure: killing every processes with SIGTERM
[WARNING] 141/114259 (12161) : All workers exited. Exiting... (134)
```

**Comment by jiangwenyuan** [comment]
@skinkie Thanks. I'll look into it.

BTW, master is for develop, please do not use master branch, use[ stable release](https://github.com/jiangwenyuan/nuster/blob/master/Download.md) instead.

**Comment by skinkie** [comment]
> BTW, master is for develop, please do not use master branch, use[ stable release](https://github.com/jiangwenyuan/nuster/blob/master/Download.md) instead.

I understand. But having a real used website run on beta code is the only way of real life validation.

**Comment by jiangwenyuan** [comment]
Do you have more info? It starts happen again in latest commit? Do you have any idea  before which commit there's no such error?

**Comment by skinkie** [comment]
This started to happen quite some while ago: https://github.com/jiangwenyuan/nuster/issues/70

**Comment by jiangwenyuan** [comment]
I see, I thought you have not encountered this error since then and just came across this again recently

**Comment by skinkie** [comment]
@jiangwenyuan I think this might be the difference between H2.1 and master. But lets see if it happens again.

**Comment by jiangwenyuan** [comment]
Thanks. I also changed one of my env from H2.1 to master to investigate.

**Comment by skinkie** [comment]
So today again.
```
[NOTICE] 141/115125 (10712) : [nuster][cache] on, dict_size=1048576, data_size=1073741824
[WARNING] 141/115125 (10712) : [./haproxy.main()] Cannot raise FD limit to 2000047, limit is 4096. This will fail in >= v2.3
[NOTICE] 141/115125 (10712) : haproxy version is 5.0.0.21-2d7ef8-119
[NOTICE] 141/115125 (10712) : path to executable is ./haproxy
[ALERT] 141/115125 (10712) : [./haproxy.main()] FD limit (4096) too low for maxconn=1000000/maxsock=2000047. Please raise 'ulimit-n' to 2000047 or more to avoid any trouble.This will fail in >= v2.3
[NOTICE] 141/115125 (10712) : New worker #1 (10713) forked
[ALERT] 142/114254 (10713) : A bogus STREAM [0x7f0b380f7ed0] is spinning at 103281 calls per second and refuses to die, aborting now! Please report this error to developers [strm=0x7f0b380f7ed0 src=86.83.61.58 fe=http be=openstreetmap_org dst=s1 rqf=48c4a000 rqa=40000 rpf=e0044060 rpa=24000000 sif=EST,208000 sib=CLO,280190 af=(nil),0 csf=0x7f0b4802a8b0,104000 ab=(nil),0 csb=0x7f0b3c0640e0,0 cof=0x7f0b380f9fc0,1300:H2(0x7f0b44243300)/SSL(0x7f0b3c059620)/tcpv4(34) cob=0x224c790,82300:H1(0x7f0b3801f460)/RAW((nil))/tcpv6(71) filters={0x7f0b440a6f40="cache filter id"}]
[ALERT] 142/114254 (10712) : Current worker #1 (10713) exited with code 134 (Aborted)
[ALERT] 142/114254 (10712) : exit-on-failure: killing every processes with SIGTERM
[WARNING] 142/114254 (10712) : All workers exited. Exiting... (134)
```

**Comment by igorescobar** [comment]
@skinkie you have a hard ulimit set to 4096. If you want to work around this you need to increase the ulimit to at least 2000047 based on your maxconn. BUT it seems that Nuster has some fd leak and @jiangwenyuan is looking into it.

**Comment by skinkie** [comment]
@igorescobar I have an issue on that system which does not allow me to do an ulimit -n unlimited (as root). But I now lowered the  maxconn sufficiently to make the error disapeer.

**Comment by skinkie** [comment]
Sadly GDB didn't give me an option to trace. But it just happened again.
```
[NOTICE] 142/123711 (4130) : [nuster][cache] on, dict_size=1048576, data_size=1073741824
[NOTICE] 142/123711 (4130) : New worker #1 (4134) forked

[NOTICE] 145/182700 (4134) : haproxy version is 5.0.0.21-2d7ef8-119
[NOTICE] 145/182700 (4134) : path to executable is /home/skinkie/nuster/haproxy
[ALERT] 145/182700 (4134) : A bogus STREAM [0xb28fb0] is spinning at 108006 calls per second and refuses to die, aborting now! Please report this error to developers [strm=0xb28fb0 src=212.92.89.85 fe=http be=openstreetmap_org dst=s1 rqf=48c4a000 rqa=40000 rpf=e0044060 rpa=24000000 sif=EST,208000 sib=CLO,280190 af=(nil),0 csf=0x7fcce42b4610,104000 ab=(nil),0 csb=0xb40800,0 cof=0x7fcce021bb20,1300:H2(0xc36b60)/SSL(0x7fcce80c2590)/tcpv4(36) cob=0x7fcce0291730,c2300:H1(0x7fcce4297ba0)/RAW((nil))/tcpv6(46) filters={0x7fcce007d9a0="cache filter id"}]
[NOTICE] 145/182700 (4130) : haproxy version is 5.0.0.21-2d7ef8-119
[NOTICE] 145/182700 (4130) : path to executable is /home/skinkie/nuster/haproxy
[ALERT] 145/182700 (4130) : Current worker #1 (4134) exited with code 134 (Aborted)
[ALERT] 145/182700 (4130) : exit-on-failure: killing every processes with SIGTERM
[WARNING] 145/182700 (4130) : All workers exited. Exiting... (134)
[Inferior 1 (process 4130) exited with code 0206]
```

**Comment by igorescobar** [comment]
@skinkie I strongly recommend you to use the latest stable version that was released today.

**Comment by skinkie** [comment]
@igorescobar switched to H2.1 now.

**Comment by jiangwenyuan** [comment]
@skinkie Thanks a lot for helping test. Strange it never happens on my env( master or H2.1), tricky bug:)

I'd invest more time if it occurs on the stable version

**Comment by jiangwenyuan** [comment]
There are several similar bugs fixed in HAProxy v2.2 and I've released v5.2.0.22 which includes all those fixes.
